### PR TITLE
Add last reclaimed segment metric to logs WAL

### PR DESCRIPTION
**Pull request by thepalbi**
#### PR Description

Small PR that adds a new metric that tacks the last reclaimed segment on the logs WAL.

#### PR Checklist

- [x] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ Main (unreleased)
   - `discovery.openstack` - service discovery for OpenStack. (@marctc)
   - `discovery.hetzner` - service discovery for Hetzner Cloud. (@marctc)
   - `loki.write` now exposes basic WAL support. (@thepalbi)
+  - `loki.write` WAL now exposes a last segment reclaimed metric. (@thepalbi)
```
Comment (tpaschalis) [inline CHANGELOG.md:48]: nit: Could we move these two log entries to "Enhancements" instead of below "New Grafana Agent Flow components".

Comment (thepalbi) [reply CHANGELOG.md:48]: Yess

Comment (tpaschalis) [reply CHANGELOG.md]: Could we also move the previous entry, since it's not a new component?

Comment (thepalbi) [reply CHANGELOG.md]: Donee

**Pull request merged**